src/darwintracelib1.0/dup2.c

```diff
@@ -68,6 +69,21 @@ static int _dt_dup2(int filedes, int filedes2) {
 		__darwintrace_sock_set(new_stream);
 	}
 
+
+    if(filedes2 == __dtsharedmemory_getStatusFileFd() || filedes2 == __dtsharedmemory_getSharedMemoryFileFd())
+    {
+        if(!__dtsharedmemory_reset_fd())
+        {
+            //if __dtsharedmemory_reset_fd() fails, do not allow overwriting either
+            //coz doing so will cause process to alter our fd and which may cause
+            //functions in dtsharedmemory.c to access offsets in memory that don't exist 
+            //causing seg faults and bus errors
+
+            fprintf(stderr, "dtsharedmemory.c : Failed to reset fd");
```

Comment:
Shouldn't this go through some darwintrace function to a logfile instead of stderr?

Reply:
I will remember to fix that too in the next commit.